test/python/circuit/library/test_qft.py

```diff
@@ -219,6 +219,27 @@ def __init__(self, *_args, **_kwargs):
                 with self.assertRaises(SentinelException):
                     qft._build()
 
+    def test_name_after_inverse_rebuild(self):
+        """Test the inverse QFT is correctly labeled, even after triggering rebuilds.
+
+        Regression test of #14758.
+        """
+        with self.assertWarns(DeprecationWarning):
+            qft = QFT(2)
+
+        iqft = qft.inverse()  # name is IQFT
+        iqft.num_qubits = 1  # name should still be IQFT, and not display IQFT_dg
```

Comment:
Do you want to put an `assert` directly on the name, as well as / instead of indirectly via the drawing?

Reply:
The funky thing is that the `.name` is actually correct -- it's the internal instruction that has the name `IQFT_dg`, which is why I went via the drawing. Alternatively we could check the name of the instruction inside the circuit.